### PR TITLE
Upgrade remoting to 2.62 and disable JNLP3 protocol by default

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It's a follow-up to the discussion in #8 

@reviewbybees (esp @carlossg and @daniel-beck )

### CODE REVIEW

File: jenkins-slave [hunk 1]
```diff
@@ -45,5 +45,10 @@ else
 		URL="-url $JENKINS_URL"
 	fi
 
-	exec java $JAVA_OPTS -cp /usr/share/jenkins/slave.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL "$@"
+	if [ ! -z "$JNLP_PROTOCOL_OPTS" ]; then
+		echo "Warning: JnlpProtocol3 is disabled by defaulr, use JNLP_PROTOCOL_OPTS to alter the behavior"
```
Comment (carlossg) [inline jenkins-slave]: typo `defaulr`

File: jenkins-slave [hunk 2]
```diff
@@ -45,5 +45,10 @@ else
 		URL="-url $JENKINS_URL"
 	fi
 
-	exec java $JAVA_OPTS -cp /usr/share/jenkins/slave.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL "$@"
+	if [ ! -z "$JNLP_PROTOCOL_OPTS" ]; then
```
Comment (carlossg) [inline jenkins-slave]: isn't this the opposite of what you want to do?  if JNLP_PROTOCOL_OPTS is not set then set it

Another option is to just add to the Dockerfile

    ENV JNLP_PROTOCOL_OPTS="-Dorg.jenkinsci.remoting.engine.JnlpProtocol3.disabled=true"

Comment (oleg-nenashev) [inline jenkins-slave]: right...

**Comment by carlossg** [comment]
:bee:

File: jenkins-slave [hunk 3]
```diff
@@ -45,5 +45,10 @@ else
 		URL="-url $JENKINS_URL"
 	fi
 
-	exec java $JAVA_OPTS -cp /usr/share/jenkins/slave.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL "$@"
+	if [ -z "$JNLP_PROTOCOL_OPTS" ]; then
+		echo "Warning: JnlpProtocol3 is disabled by default, use JNLP_PROTOCOL_OPTS to alter the behavior"
```
Comment (daniel-beck) [inline jenkins-slave]: Should this really be a "warning"?

**Comment by daniel-beck** [comment]
Other than the "warning" term used, :+1:

**Comment by oleg-nenashev** [comment]
@carlossg Do you maintain the images now? If yes, could you please release it?

**Comment by carlossg** [comment]
@oleg-nenashev master, branches and tags are automatically built, do you want to merge this PR?
there are some comments from @daniel-beck

**Comment by oleg-nenashev** [comment]
Yes, I think warning is valid, because it notifies people that JNLP3 with security features is disabled by default

**Comment by jglick** [comment]
@oleg-nenashev @jeffret-b et al., can the `-Dorg.jenkinsci.remoting.engine.JnlpProtocol3.disabled=true` be removed at this point, given that this protocol has been removed from the server side anyway? Just makes for noise in JVM command lines.

**Comment by jeffret-b** [comment]
I can't see any reason to keep it.

**Comment by oleg-nenashev** [comment]
Agreed

**Pull request merged**